samples/03.live-inking/src/main-view.ts

```diff
@@ -0,0 +1,334 @@
+
+            this._inkingManager = new InkingManager(inkingHost);
+
+            await inkingSession.initialize(this._inkingManager);
+
+            this._inkingManager.activate();
+
+            this._drawingSimulation = new DrawingSimulation(this._inkingManager);
+
+            this._hostResizeObserver = new ResizeObserver(() => { this.updateBackgroundImagePosition(); });
+            this._hostResizeObserver.observe(inkingHost);
+
+            /*
+            // Set which roles can draw on the canvas. By default, all roles are allowed
+            inkingSession.allowedRoles = [ UserMeetingRole.presenter ];
+            */
+        }
+
+        this.updateBackgroundImagePosition();
+    }
+
+    private _backgroundImageWidth?: number;
+    private _backgroundImageHeight?: number;
+
+    private updateBackgroundImagePosition() {
+        const backgroundImage = document.getElementById("backgroundImage");
+
+        if (backgroundImage && this._inkingManager && this._backgroundImageWidth && this._backgroundImageHeight) {
+            backgroundImage.style.removeProperty("visibility");
+
+            const actualWidth = this._backgroundImageWidth * this._inkingManager.scale;
+            const actualHeight = this._backgroundImageHeight * this._inkingManager.scale;
+
+            backgroundImage.style.width = actualWidth + "px";
+            backgroundImage.style.height = actualHeight + "px";
+            backgroundImage.style.left = (this._inkingManager.centerX + this._inkingManager.offset.x - this._backgroundImageWidth / 2 * this._inkingManager.scale) + "px";
+            backgroundImage.style.top = (this._inkingManager.centerY + this._inkingManager.offset.y - this._backgroundImageHeight / 2 * this._inkingManager.scale) + "px";
+        }
+    }
+
+    constructor() {
+        super();
+
+        this._userInfo = getRandomUserInfo();
+
+        Utils.loadTemplate(appTemplate, document.body);
+
+        const backgroundImage = document.getElementById("backgroundImage") as HTMLImageElement;
+
+        if (backgroundImage) {
+            const showBackgroundImage = () => {
+                this._backgroundImageWidth = backgroundImage.naturalWidth;
+                this._backgroundImageHeight = backgroundImage.naturalHeight;
+
+                this.updateBackgroundImagePosition();
+            }
+
+            if (backgroundImage.complete) {
+                showBackgroundImage();
+            }
+            else {
+                backgroundImage.addEventListener(
+                    "load",
+                    () => {
+                        showBackgroundImage();
+                    });
+            }
+        }
+
+        const setupButton = (buttonId: string, onClick: () => void) => {
+            const button = document.getElementById(buttonId);
+
+            if (button) {
+                button.onclick = onClick;
+            }
+        }
+
+        setupButton("btnStroke", () => { this._inkingManager.tool = InkingTool.pen });
```

Comment:
nit: could this be combined with the ink color changing, so that user only needs to press 1 button if they are switching from (e.g.) highlighter to pen?